node_modules/oae-activity/lib/internal/dao.js

```diff
@@ -256,23 +256,29 @@ var getQueuedActivities = module.exports.getQueuedActivities = function(bucketNu
             return callback(err);
         }
 
-        // The Redis result is each value on a new line, in order of "rank" in the sorted-list. Iterate over those and parse the values in order.
+        var numToDelete = _.size(routedActivities);
+
+        // The Redis result is each value on a new line, in order of "rank" in
+        // the sorted-list. Iterate over those and parse the values in order
         var queuedActivities = {};
         _.each(routedActivities, function(routedActivity) {
             try {
                 // Routed activities are stored as stringified JSON, so we parse them back to objects
                 routedActivity = JSON.parse(routedActivity);
+                queuedActivities[_createRoutedActivityKey(routedActivity)] = routedActivity;
             } catch (err) {
-                log().warn({'err': err, 'routedActivity': routedActivity}, 'Error trying to parse stored routed activity.');
-                return;
-            }
+                var truncatedRoutedActivity = JSON.stringify(routedActivity);
+                if (truncatedRoutedActivity) {
+                    truncatedRoutedActivity = truncatedRoutedActivity.slice(0, 300);
+                }
 
-            queuedActivities[_createRoutedActivityKey(routedActivity)] = routedActivity;
+                log().error({'err': err, 'routedActivity': truncatedRoutedActivity}, 'Error trying to parse stored routed activity');
+            }
         });
 
-        log().trace({'queuedActivities': queuedActivities}, 'Fetched queued activities.');
+        log().trace({'queuedActivities': queuedActivities}, 'Fetched queued activities');
 
-        return callback(null, queuedActivities);
+        return callback(null, queuedActivities, numToDelete);
```

Comment:
That's intentional, because we want to delete items that can't be parsed, so they should be included in the "num to delete" count.